### PR TITLE
README: update the branch usage

### PR DESCRIPTION
**Pull request by Vicente-Cheng**
Add some branch descriptions on README.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -3,6 +3,12 @@ Harvester Node Manager
 
 A node manager helps to manage the host kernel configuration of the [Harvester](https://github.com/harvester/harvester) cluster, eg: KSM.
 
+## Branchs
```
Comment (connorkuehl) [inline README.md:6]:
```suggestion
## Branches
```

**Review by connorkuehl — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (Vicente-Cheng) [reply README.md]: updated, thanks!

File: README.md [hunk 2]
```diff
@@ -3,6 +3,12 @@ Harvester Node Manager
 
 A node manager helps to manage the host kernel configuration of the [Harvester](https://github.com/harvester/harvester) cluster, eg: KSM.
 
+## Branches
+
+- `master` branch is used for development and release.
+- `v0.1.x` branch is used for Harvester v1.2.x release.
```
Comment (innobead) [inline README.md:9]: Would it be possible to align the branch name with the Harvester repo in the future? This would make it more intuitive.

**Review by bk201 — APPROVED** [review]

File: README.md (hunk 2 above)
Comment (bk201) [reply README.md:9]: I'd suggest we do this across projects in a future milestone, but still need to check if there are any potential issues of inconvenience. There are some projects like harveter-csi-dirver, cloud-provider, terraform are not bound to harvester versions.

Comment (innobead) [reply README.md:9]: Sounds good. For those projects not bundled to harvester release, then individual release is reasonable.

Comment (Vicente-Cheng) [reply README.md:9]: I created a GH issue https://github.com/harvester/harvester/issues/5559 to track this.

I will merge this PR first, and we can discuss the above issue.

**Pull request merged**